app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -44,6 +44,19 @@ def save_ems_infra_inventory(ems, hashes, target = nil)
       return
     end
 
+    # Check if a vm was removed and we still have disk(s)
+    if target.kind_of?(VmOrTemplate) && hashes[:vms].blank?
+      current = hashes[:vm_disks].collect { |disk| URI(disk[:storage][:ems_ref]).path.split('/').last }.uniq
```

Comment:
I don't think we can assume here that the `ems_ref` is going to be parseable as a `URI`.  I believe that this will only be true for RHEV.

Is there another way you can do the comparisons below that doesn't involve parsing the `ems_ref` like this?